component-samples/http-device-sample/src/main/java/org/fido/iot/sample/DeviceSysModule.java

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020 Intel Corporation
+// SPDX-License-Identifier: Apache 2.0
+
+package org.fido.iot.sample;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.FileSystems;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+import org.fido.iot.protocol.Composite;
+import org.fido.iot.protocol.Const;
+import org.fido.iot.serviceinfo.FdoSys;
+import org.fido.iot.serviceinfo.Module;
+
+/**
+ * Implements the device side of the FdoSys Module.
+ */
+public class DeviceSysModule implements Module {
+
+  // Keys for localized log format strings in our resource bundle
+  private static final String LOG_ERR_PATH_NULL =
+      DeviceSysModule.class.getName() + ".LOG_ERR_PATH_NULL";
+  private static final String LOG_INFO_FILE_CREATED =
+      DeviceSysModule.class.getName() + ".LOG_INFO_FILE_CREATED";
+
+
+  private ProcessBuilder.Redirect execOutputRedirect = ProcessBuilder.Redirect.PIPE;
+  private Duration execTimeout = Duration.ofHours(2);
+  private Predicate<Integer> exitValueTest = val -> (0 == val);
+
+  private Path currentFile;
+  private boolean isActive;
+
+  @Override
+  public String getName() {
+    return FdoSys.NAME;
+  }
+
+  @Override
+  public void prepare(UUID guid) {
+    currentFile = null;
+    isActive = false;
+  }
+
+  @Override
+  public void setMtu(int mtu) {
+
+  }
+
+  @Override
+  public void setState(Composite state) {
+
+  }
+
+  @Override
+  public Composite getState() {
+    return null;
+  }
+
+  @Override
+  public void setServiceInfo(Composite kvPair, boolean isMore) {
+    String name = kvPair.getAsString(Const.FIRST_KEY);
+    switch (name) {
+      case FdoSys.KEY_ACTIVE:
+        isActive = kvPair.getAsBoolean(Const.SECOND_KEY);
+        break;
+      case FdoSys.KEY_FILEDESC:
+        createFile(Path.of(kvPair.getAsString(Const.SECOND_KEY)));
+        break;
+      case FdoSys.KEY_WRITE:
+        writeFile(kvPair.getAsBytes(Const.SECOND_KEY));
+        break;
+      case FdoSys.KEY_EXEC:
+        exec(kvPair.getAsComposite(Const.SECOND_KEY));
+        break;
+      default:
+        break;
```

Comment:
Instead of silently skipping unknown keys, should we be logging an error or throwing an exception?